### PR TITLE
[ncurses] Update compilation option of 6.2 release

### PR DESCRIPTION
**Pull request by vincowl**
Relates to #3632

### CODE REVIEW

File: packages/ncurses.rb [hunk 1]
```diff
@@ -3,56 +3,47 @@
...
              "--libdir=#{CREW_LIB_PREFIX}",
-             "--without-normal",
+             "--with-normal",
              "--with-shared",
              "--with-cxx-shared",
              "--without-debug",
              "--enable-pc-files",
              "--with-pkg-config-libdir=#{CREW_LIB_PREFIX}/pkgconfig",
-             "--enable-widec"
+             "--without-manpages",
```
Comment (uberhacker) [inline packages/ncurses.rb]: Why without manpages?

Comment (vincowl) [reply packages/ncurses.rb]: I inverted the 2 builds. This option was not set by me. Indeed, why not actvating man pages... Should we set it to --with-manpages ?

Comment (uberhacker) [reply packages/ncurses.rb]: Good question.  It might just build the manpages by default unless `--without-manpages` is provided.

**Comment by uberhacker** [comment]
I have the sinking feeling this is going to break some packages that already depend on ncurses and there are A LOT of packages that have this dependency.  We need to thoroughly test this also since it's a core package and requires the binaries to install Chromebrew itself.

**Comment by cstrouse** [comment]
@uberhacker Yeah, but there's a lot of things where I can't build from source on arm related to ncurses/terminfo. I'll help test things with the containers. If you put together a list of the most critical testing that needs to happen I'll script it for us.

File: packages/ncurses.rb (hunk 1 above)
Comment (cstrouse) [reply packages/ncurses.rb]: It'd be nice to set that globally and have everything build docs or none of them. The Android stuff eats up so much of my space on the Samsung Chromebook Plus v1 that if I build certain packages it runs out of space and starts deleting things. An option to slim things down would be nice.

We could also take the position that Crux does and just get rid of everything that's not essential to the package functioning.

**Comment by uberhacker** [comment]
These are the kinds of things I test when I do it manually:
- Identify the binaries from the package, if any, and attempt to execute (something like <command> -h or <command> -v).
- If there are any libraries, make sure they are installed in the correct #{CREW_LIB_PREFIX} directory.
- If there are any manpages, attempt to run man <command> or info <command>.  If not working, check if they are installed in #{CREW_PREFIX}/man instead of the preferred #{CREW_PREFIX}/share/man.
- Execute crew files <package> to make sure files exist and it looks reasonable (count > 1 and no missing file messages).
- If it is a dependency of other packages, test to make sure the binaries of the other packages still work as expected.
- Uninstall the package and check for any residue left behind.  Removing the package should restore to a state exactly is it was before the package was installed.

**Comment by cstrouse** [comment]
All of the packages that are using `#{CREW_PREFIX}/include/ncursesw` as CPPFLAGS (like `sngrep`) are now broken.

File: packages/ncurses.rb (hunk 1 above)
Comment (vincowl) [reply packages/ncurses.rb]: I would avoid adding manpages to save space but it remains a personal choice. Adding a global option would be great indeed, but out of this package's scope

**Comment by vincowl** [comment]
It's really late here to do it by myself. Maybe try to just keep previous compilation options and juste add --with-termlib. That should be enough to make hunspell run correctly, and maybe avoid other packages crashing. Indeed, documentation states that termlib is a decorrelated part

File: packages/ncurses.rb (hunk 1 above)
Comment (uberhacker) [reply packages/ncurses.rb]: How about a separate package called `ncurses_manpages.rb`?  Other package managers separate out like this for manpages that require a lot of space.

**Comment by vincowl** [comment]
I didn't try to test the new hunspell package with old ncurses (without this PR). Maybe it's enough and would delay this debate on a possible huge impact while having a working hunspell version

**Comment by uberhacker** [comment]
The testing scope of this change is definitely greater than with most other packages.  We always need to be careful when updating libraries because the version changes tend to break things spectacularly.

**Comment by uberhacker** [comment]
@cstrouse: I added the manual testing procedure [here](https://github.com/skycocker/chromebrew/wiki/Testing-procedure).  You can add some scripts here (or links to the tools directory) if you like also.

**Comment by JL2210** [comment]
@uberhacker ncurses and ncursesw are ABI-compatible.

**Comment by vincowl** [comment]
I performed several tests. We could just add the `--with-libterm` argument and keep previous parameters as is. But I don't understand what happened yesterday but it implies to apply the patch of configure.ac (lines to modifiy or add) we get once the hunspell install archive is extracted. Don't know how to apply a patch sinceit implies to download a patch we would host and maintain, no idea to do it easily with sed since it implies lines addition or modifications. We could use a cat configure.ac | tr '\n' '#' | sed .........| tr '#' '\n' ` But this line will be REALLY ugly....

**Comment by vincowl** [comment]
What shall we do here ? Actually I added these modifications to solve hunspell troubles. But other PRs did the trick. Maybe we could close this one and avoid overthinking ?

**Comment by uberhacker** [comment]
Closing but the next time this package has an update, we need to consider what is mentioned in the comments here.

**Comment by vincowl** [comment]
So maybe we should decide on the cases raised here. Maybe in new issues linked to this PR ?

**Pull request closed without merging**